### ISSUE TITLE
Fix feature flag usage

### ISSUE THREAD

**Issue by dferber90** [haydenbleasel/next-forge#175]
Lovely template! I noticed some unnecessary code with how feature flags are set up [here](https://github.com/haydenbleasel/next-forge/blob/e0a5c2d7bfc53bcc50295280b200b7c7f549151d/packages/feature-flags/lib/create-flag.ts#L10). The flag created by `unstable_flag` will automatically respect the overrides you set up, so you can delete the lines related to handling the override manually.

You also don't need the try/catch block if you specify a defaultValue, as the `decide` function automatically catches and returns the defaultValue if one is set. Further, you can remove the hardcoded `false` and instead refer to `this.defaultValue`.

Here's an improved version:

```ts
export const createFlag = (key: string) => {
  return flag({
    key,
   defaultValue: false,
    async decide(params) {
        const { userId } = await auth();
        if (!userId) return this.defaultValue;
        const isEnabled = await analytics.isFeatureEnabled(key, userId);
        return isEnabled ?? this.defaultValue;
    },
  });
};
```

**Comment by haydenbleasel** [comment]
Nice, thanks for that! I didn't know about the automatic overrides so that sounds good.

In relation to the try / catch block, it's there as you may not be running this inside a ClerkProvider, in which case I was seeing it crash on the web project.

I could be wrong though, I'll double check.

**Comment by dferber90** [comment]
I just published v2.6.3 of our `@vercel/flags` SDK which includes a bug fix. We now catch when an async `decide` function throws and apply the `defaultValue` if it was specified. This is what I was referring to with not having to use the try/catch.
